### PR TITLE
CON-2441 Added URL fetching on app start and cart page entering

### PR DESCRIPTION
**Pull request by richardgorman**

### CODE REVIEW

File: cart/streams/index.js [hunk 1]
```diff
@@ -97,6 +102,13 @@ export const couponPushNotification$ = openedLink$
   .filter(
     ({ action }) => {
       /**
+       * Stop here if we didn't receive a url.
```
Comment (devbucket) [inline cart/streams/index.js]: Why block comment? Please, make it a single line comment: `// ...`

File: cart/streams/index.js [hunk 2]
```diff
@@ -97,6 +102,13 @@ export const couponPushNotification$ = openedLink$
...
+       */
+      if (!action.options || !action.options.url) {
+        return false;
+      }
+
+      /**
        * Split the URL string and remove any empty strings.
```
Comment (devbucket) [inline cart/streams/index.js]: Why block comment? Please, make it a single line comment: `// ...`

File: cart/subscriptions/index.js [hunk 3]
```diff
@@ -41,6 +44,8 @@ export default function cart(subscribe) {
    */
   const cartNeedsSync$ = userDidUpdate$.merge(remoteCartDidUpdate$);
 
+  const cartDidEnterOrAppDidStart$ = routeDidEnter(CART_PATH).merge(appDidStart$);
```
Comment (devbucket) [inline cart/subscriptions/index.js]: Maybe add a comment to describe what the stream is for, please

File: cart/streams/index.js (hunk 1 above)
Comment (richardgorman) [reply cart/streams/index.js]: Changed

File: cart/streams/index.js (hunk 2 above)
Comment (richardgorman) [reply cart/streams/index.js]: Changed

File: cart/subscriptions/index.js (hunk 3 above)
Comment (richardgorman) [reply cart/subscriptions/index.js]: Added a comment

**Pull request merged**